src/cache.gateway.ts

```diff
@@ -0,0 +1,27 @@
+
+import type { Cache as GatewayCache } from '@adaline/gateway';
```

Comment:
Maybe we can tuck this into `providers` subdirectory as it is a provider-specific cache?